### PR TITLE
Feature/navigator component icons multifile

### PR DESCRIPTION
**Pull request by enidemi**
**Problem:**
Navigator icons are not aware if something is focusable or not since multifile editing. And Context-menu enable/disable state will need the same functionality.

**Commit Details:**
- allow context-menu items enable/disable themselves from a callback
- added helper `isFocusableComponent` that will be used later for the context-menu, now it is used for the navigator component icons
- added helper that returns components and imports for instances, like for sample project card-instance will have the `app.js` components
- added helper that returns components for instances + original components and original imports, for the sample project card-instance will have the `card.js` components and imports + `app.js` components. in this version the `app.js` components are needed because otherwise the elementname for path resolve is not working, I think the extra components shouldn't cause problems.

### CODE REVIEW

File: editor/src/components/context-menu-items.ts [hunk 1]
```diff
@@ -135,9 +135,10 @@ export const toggleShadowItem: ContextMenuItem<CanvasData> = {
 }
 
 export const setAsFocusedElement: ContextMenuItem<CanvasData> = {
-  name: 'Edit Component',
-  // todo only enable if thing is component
-  enabled: true,
+  name: 'Set As Focused Element',
+  enabled: (data) => {
+    return true
```
Comment (balazsbajorics) [inline editor/src/components/context-menu-items.ts:140]: I think this is still not going to be good enough here :)

Comment (enidemi) [reply editor/src/components/context-menu-items.ts:140]: that's for the next PR! :D I should have split it into 2 branches :(

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/core/model/element-metadata-utils.ts [hunk 2]
```diff
@@ -1539,6 +1545,36 @@ export const MetadataUtils = {
...
+    path: TemplatePath,
+    components: UtopiaJSXComponent[],
+    metadata: ElementInstanceMetadataMap,
+    imports: Imports,
+  ): boolean {
+    const elementName = MetadataUtils.getJSXElementName(path, components, metadata)
+    const elementInstancePath = TP.isInstancePath(path)
+      ? path
+      : TP.instancePathForElementAtScenePath(path)
+    const element = MetadataUtils.getElementByInstancePathMaybe(metadata, elementInstancePath)
```
Comment (Rheeseyb) [inline editor/src/core/model/element-metadata-utils.ts:1558]: We have a function for doing exactly this `findElementByTemplatePath`, which exists for smoothing the transition between the separate path types and a future where they don't exist

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/core/model/element-metadata-utils.ts (hunk 2 above)
Comment (enidemi) [reply editor/src/core/model/element-metadata-utils.ts]: fixed!

**Pull request merged**